File: Backend-SpringBoot/FoodApp/src/main/java/com/tientien/foodapp/order/dto/CheckoutRequest.java
package com.tientien.foodapp.order.dto;

import com.tientien.foodapp.common.enums.PaymentMethod;
import lombok.Data;

@Data
public class CheckoutRequest {
    private Long userId;
    private String shippingAddress;
    private String phoneNumber;
    private String note;
    private PaymentMethod paymentMethod;
    private String voucherCode;
}

File: Frontend_Flutter/lib/models/checkout_request.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'checkout_request.freezed.dart';
part 'checkout_request.g.dart';

@freezed
class CheckoutRequest with _$CheckoutRequest {
  const factory CheckoutRequest({
    required int userId,
    required String shippingAddress,
    required String phoneNumber,
    String? note,
    String? paymentMethod, // COD, VNPAY, MOMO, CARD
    String? voucherCode,
  }) = _CheckoutRequest;

  factory CheckoutRequest.fromJson(Map<String, dynamic> json) =>
      _$CheckoutRequestFromJson(json);
}
